OWML.ModHelper.Input/ModInputHandler.cs

```diff
@@ -0,0 +1,375 @@
+﻿using System;
+using System.Reflection;
+using System.Collections.Generic;
+using OWML.Common;
+using UnityEngine;
+
+namespace OWML.ModHelper.Input
+{
+    public class ModInputHandler : IModInputHandler
+    {
+        internal static ModInputHandler Instance { get; private set; }
+
+        private HashSet<IModInputCombination> _singlePressed = new HashSet<IModInputCombination>();
+        private Dictionary<long, IModInputCombination> _comboRegistry = new Dictionary<long, IModInputCombination>();
+        private HashSet<InputCommand> _gameBindingRegistry = new HashSet<InputCommand>();
+        private float[] _timeout = new float[_maxUsefulKey];
+        private int[] _gameBindingCounter = new int[_maxUsefulKey];
+        private ModInputCombination _lastPressed;
+        private float _lastUpdate;
+        private float _lastSingleUpdate;
+        private static readonly float _updateCooldown = 0.01f; //aka little bit less than one frame
```

Comment:
Is it super important that this is less than one frame? Because at higher framerates that is no longer the case (and mods can unlock higher framerates)

Reply:
actually, I should probably replace it with frametime thing

Reply:
done, now it's UnscaledDeltaTime